src/workerd/jsg/value.h

```diff
@@ -1021,17 +1045,22 @@ class DictWrapper {
     auto object = handle.As<v8::Object>();
     v8::Local<v8::Array> names = check(object->GetOwnPropertyNames(context));
     auto length = names->Length();
-    auto builder = kj::heapArrayBuilder<typename Dict<V, K>::Field>(length);
+
+    // Use a HashMap during construction because the conversion to the string type K could
```

Comment:
This adds a bunch of overhead, and in the common case that the key type `K` is `kj::String` then this overhead is completely wasted. We need to gate this to apply only when the type is `USVString`. (Even then, I'm really dubious about whether anyone cares about this.)

Reply:
Definitely agree with Kenton here. Less so about whether anyone cares about it or not but about the added overhead the extra book keeping here adds.

Reply:
Given that we're not sure we want to ignore duplicate keys, I backed out the changes to Dict - it'll just produce a list of pairs and callers can decide what to do.